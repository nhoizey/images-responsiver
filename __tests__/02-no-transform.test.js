'use strict';

const prettier = require('prettier');
const imagesResponsiver = require('../index');

function cleanHtml(html) {
	return prettier.format(html, { parser: 'html' });
}

describe("image that can't be transformed", () => {
	test('do nothing on SVG image', () => {
		const content = `<!DOCTYPE html><html><body><img src="test.svg"></body></html>`;
		const transformed = imagesResponsiver(content);
		expect(cleanHtml(transformed)).toMatchSnapshot();
	});

	test('do nothing on image without a src', () => {
		const content = `<!DOCTYPE html><html><body><img alt="not really an image"></body></html>`;
		const transformed = imagesResponsiver(content);
		expect(cleanHtml(transformed)).toMatchSnapshot();
	});

	test('do nothing on image with already a srcset', () => {
		const content = `<!DOCTYPE html><html><body><img src="test.png" srcset="test.png 320w"></body></html>`;
		const transformed = imagesResponsiver(content);
		expect(cleanHtml(transformed)).toMatchSnapshot();
	});

	test('do nothing on image with data-responsiver="false"', () => {
		const content = `<!DOCTYPE html><html><body><img src="test.png" data-responsiver="false"></body></html>`;
		const transformed = imagesResponsiver(content);
		expect(cleanHtml(transformed)).toMatchSnapshot();
	});
});
