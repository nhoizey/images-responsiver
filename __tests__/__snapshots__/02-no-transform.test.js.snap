// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`image that can't be transformed do nothing on SVG image 1`] = `
"<!DOCTYPE html>
<html>
  <body>
    <img src=\\"test.svg\\" />
  </body>
</html>
"
`;

exports[`image that can't be transformed do nothing on image with already a srcset 1`] = `
"<!DOCTYPE html>
<html>
  <body>
    <img src=\\"test.png\\" srcset=\\"test.png 320w\\" />
  </body>
</html>
"
`;

exports[`image that can't be transformed do nothing on image with data-responsiver="false" 1`] = `
"<!DOCTYPE html>
<html>
  <body>
    <img src=\\"test.png\\" data-responsiver=\\"false\\" />
  </body>
</html>
"
`;

exports[`image that can't be transformed do nothing on image without a src 1`] = `
"<!DOCTYPE html>
<html>
  <body>
    <img alt=\\"not really an image\\" />
  </body>
</html>
"
`;
